src/sagas/repository.js

```diff
@@ -13,13 +14,49 @@ import { ActionTypes } from 'constants/index';
  * @param {Object} action
  *
  */
+
+/**
+ * Saga poller
+ */
+function* pollJobWorker(jobId) {
+  try {
+    const response = yield call(axios.get, '/api/repository/v1/jobs/' + jobId);
+    const jobStatus = response.data.job_status;
+    if (jobStatus !== 'running') {
+      const resultResponse = yield call(axios.get, '/api/repository/v1/jobs/' + jobId + '/result');
+      yield put({
+        type: ActionTypes.GET_JOB_RESULT_SUCCESS,
+        payload: { jobResult: resultResponse.data },
+      });
+    } else {
+      yield put({
+        type: ActionTypes.GET_JOB_BY_ID_SUCCESS,
```

Comment:
We should distinguish between success and failure here -- this branch will be reached by both cases (I think it should be two different actions)